### PR TITLE
feat: add import cluster configure settings support

### PR DESCRIPTION
**Pull request by assafgi**

### CODE REVIEW

File: internal/aws/cluster/apigateway.go [hunk 1]
```diff
@@ -76,9 +78,10 @@ func (a *ApiGateway) Delete() error {
 	return apigateway.DeleteRestApiGateway(a.ResourceName())
 }
 
-func (a *ApiGateway) Create(tags cluster.Tags, privateSubnet bool) error {
+func (a *ApiGateway) Create(tags cluster.Tags) error {
 	vpcId := ""
-	if privateSubnet {
+	if a.ClusterSettings.PrivateSubnet {
+		// TODO: Save it in settings. Cluster update will need to validate that settings have the information and fetch/update one-time in case it's missing, for back compatibility
```
Comment (tigrawap) [inline internal/aws/cluster/apigateway.go:84]: By IT I mean vpcId - can be fetched one time on import, saved in DB and then reused to save the call. 
Forst update will not have it, and therefore will need to fix it

File: internal/cluster/types.go [hunk 2]
```diff
@@ -1,9 +1,36 @@
...
 type IClusterSettings interface {
 	Tags() Tags
 	UsePrivateSubnet() bool
 }
+
+type ImportParams struct {
+	Name          string
+	Username      string
+	Password      string
+	TagsList          []string
```
Comment (assafgi) [inline internal/cluster/types.go:20]: indentation...

File: internal/aws/cluster/apigateway.go [hunk 3]
```diff
@@ -105,8 +108,7 @@ func (a *ApiGateway) Update() error {
 			return err
 		}
 		//TODO:implement fetch PrivateSubnet
```
Comment (assafgi) [inline internal/aws/cluster/apigateway.go:110]: you can remove the TODO...

File: internal/aws/cluster/common.go [hunk 4]
```diff
@@ -21,35 +25,35 @@ func GenerateHostGroup(clusterName cluster.ClusterName, hostGroupParams common.H
...
 	}
 
-	//TODO: This might not be single one
 	clientsHostGroup, err := generateHostGroupFromLaunchTemplate(
 		name, common.RoleClient, "Clients")
 	if err != nil {
 		return
 	}
 
+	hostGroups := append(backendsHostGroup, clientsHostGroup...)
```
Comment (tigrawap) [inline internal/aws/cluster/common.go:49]: backend does not have to be single one

File: internal/aws/cluster/update.go [hunk 5]
```diff
@@ -3,58 +3,70 @@ package cluster
...
+		}
+		nextToken = asgOutput.NextToken
 	}
-
 	return
 }
 
-func generateHostGroupFromLaunchTemplate(clusterName cluster.ClusterName, role common.InstanceRole, name common.HostGroupName) (hostGroup HostGroup, err error) {
-	hostGroupParams, err := fetchClusterLaunchTemplateParams(clusterName, name)
+func generateHostGroupFromLaunchTemplate(clusterName cluster.ClusterName, role common.InstanceRole, name common.HostGroupName) (hostGroups []HostGroup, err error) {
```
Comment (tigrawap) [inline internal/aws/cluster/update.go:62]: ok, function name is not right now and that's what confused me in above comment.
it is basically GetHostGroupsByRole

**Pull request merged**